src/gpu/generic/sycl/ref_softmax.hpp

```diff
@@ -48,6 +48,7 @@ struct ref_sycl_softmax_fwd_t : public gpu::generic::sycl::primitive_t {
                     && sycl_post_ops_t::post_ops_ok(attr(), true, false)
                     && set_default_formats() == status::success
                     && attr_.set_default_formats(dst_md()) == status::success
+                    && check_formats(src_md(), dst_md())
```

Comment:
I believe in the backward pass we should check `diff_src`, `diff_dst` & `dst`, `src` is not used in the backward pass.

Reply:
fixed